spec/date_version_calculator_spec.rb

```diff
@@ -16,8 +16,8 @@
         it 'increments the major version number and sets the minor version number to 1' do
           allow(Time).to receive(:now).and_return(Time.new(2023, 12, 3))
           allow(FastlaneCore::UI).to receive(:confirm).and_return(true)
-          version = Fastlane::Models::AppVersion.new('1999.30.1.2')
-          bumped_version = described_class.new.calculate_next_release_version(version)
+          version = Fastlane::Models::AppVersion.new(1999, 30, 1, 2)
+          bumped_version = described_class.new.next_release_version(after: version)
           expect(bumped_version.to_s).to eq('2000.1.0.0')
         end
```

Comment:
Sorry for the confusion 😃 The current time is mocked to December because the logic in the `date_calculator` class prompts the user if it's the first release of next year only when it's December. Because the date-versioned apps like Day One release every two weeks, there should be ~26 main releases a year (hotfixes still just do a patch x.x.1 off the main release version) but there may be some variation + or - 1 - 2 depending on how the schedule shakes out. Because of that, there's not really a programmatic way to determine whether 2023.26 is the last release of the year, or whether 2023.25/2023.27 are instead. Like I said in https://github.com/wordpress-mobile/release-toolkit/pull/512/#pullrequestreview-1629727531, this won't work with release management in CI, but Jeremy was okay with leaving it as-is for now until we figure out a better solution.